src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -131,11 +130,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
   TiffEntry* offsets = raw->getEntry(STRIPOFFSETS);
   TiffEntry* counts = raw->getEntry(STRIPBYTECOUNTS);
 
-  LJpegPlain l(*mFile, offsets->getInt(), counts->getInt(), mRaw);
-  l.addSlices(s_width);
+  Cr2Decompressor d(*mFile, offsets->getInt(), counts->getInt(), mRaw);
```

Comment:
Hm, I just had a look at the new `JpegDecompressor` and `DeflateDecompressor` and I think the design to wrap the decode function into such a class makes the code more complicated that helpful. The decode function needs a couple parameters. Now some are passed to the constructor of the class, some are passed to the member function. This is arbitrary. Also the use case is the same every time:
```
   SomeDecompressor d(some, parameters);
   d.decode( some, more, parameters ); // executed exactly once
   // destroy d
```
For the `Decoders` it is something completely different, because you really have an abstraction there: the client gets a pointer to some decoder implementation and the type `RawDecoder` does not know which it is. With the decompressors, everything is fully known.

I suggest to model this functionality as a dead simple function call. See `NikonDecompressor`.